src/cdk/scrolling/virtual-scroll-viewport.ts

```diff
@@ -270,13 +278,20 @@ export class CdkVirtualScrollViewport implements OnInit, OnDestroy {
   }
 
   /**
-   * Scrolls to the offset on the viewport.
+   * Scrolls to the given offset from the start of the viewport. Please note that this is not always
+   * the same as setting `scrollTop` or `scrollLeft`. In a horizontal viewport with right-to-left
+   * direction, this would be the equivalent of setting a fictional `scrollRight` property.
    * @param offset The offset to scroll to.
    * @param behavior The ScrollBehavior to use when scrolling. Default is behavior is `auto`.
    */
   scrollToOffset(offset: number, behavior: ScrollBehavior = 'auto') {
     const viewportElement = this.elementRef.nativeElement;
 
+    // For a horizontal viewport in a right-to-left language we need to calculate what `scrollRight`
+    // would be.
+    offset = this.orientation == 'horizontal' && this._dir && this._dir.value == 'rtl' ?
```

Comment:
It looks like you're repeating `Math.max(0, this._totalContentSize - this._viewportSize - offset)` at least in a couple of places. Consider moving it into a helper.